Google.PowerShell/Compute/GceInstanceCmdlets.cs

```diff
@@ -1101,14 +1201,10 @@ private void ProcessDisk()
             foreach (object diskParam in AddDisk)
             {
                 // Allow for taking Disk, AttachedDisk, and string objects.
-                AttachedDisk newDisk;
-                if (diskParam is AttachedDisk || (diskParam as PSObject)?.BaseObject is AttachedDisk)
-                {
-                    newDisk = diskParam as AttachedDisk ?? (diskParam as PSObject)?.BaseObject as AttachedDisk;
-                }
-                else
+                var newDisk = diskParam as AttachedDisk ?? (diskParam as PSObject)?.BaseObject as AttachedDisk;
+                if (newDisk == null)
                 {
-                    Disk disk = diskParam as Disk ?? (diskParam as PSObject)?.BaseObject as Disk;
+                    var disk = diskParam as Disk ?? (diskParam as PSObject)?.BaseObject as Disk;
```

Comment:
Should we tease this out into a method?

`Common.CoerceParameter(diskParam, typeof(Disk))`

The `as PSObject)?.` seems pretty sketch to copy and paste around a lot.